CMSIS/NN/Tests/UnitTest/generate_test_data.py

```diff
@@ -513,35 +585,44 @@ def max_pooling(self, x):
 
     args = parse_args()
 
-    if args.type == 'conv':
-        generator = ConvSettings(args, in_ch=4, out_ch=17, x_in=15, y_in=15, w_x=1, w_y=1, stride_x=1, stride_y=1,
-                                 pad=False, randmin=1, randmax=4, outminrange=-126, outmaxrange=127)
-        #conv_3
+    if args.type == 'conv' or args.type == 'depthwise_conv':
+        # kernel1x1
+        # generator = ConvSettings(args, in_ch=4, out_ch=17, x_in=15, y_in=15, w_x=1, w_y=1, stride_x=1, stride_y=1,
+        #                         pad=False, randmin=1, randmax=4, outminrange=-126, outmaxrange=127)
+
+        # conv_3
         # generator = ConvSettings(args, in_ch=3, out_ch=1, x_in=10, y_in=49, w_x=4, w_y=10, stride_x=1, stride_y=2,
         #                          pad=True, randmin=-2, randmax=2, outminrange=-127, outmaxrange=127)
 
-        #conv_1_x_n_1
+        # conv_1_x_n_1
         # generator = ConvSettings(args, in_ch=3, out_ch=3, x_in=5, y_in=5, w_x=2, w_y=1, stride_x=2, stride_y=1,
         #                          pad=False, randmin=-2, randmax=2, outminrange=-127, outmaxrange=127, batches=2)
 
-        #conv_1_x_n_2
+        # conv_1_x_n_2
         # generator = ConvSettings(args, in_ch=3, out_ch=1, x_in=11, y_in=11, w_x=11, w_y=1, stride_x=1, stride_y=1,
         #                          pad=True, randmin=-2, randmax=2, outminrange=-127, outmaxrange=127)
 
-        #conv_1_x_n_3
+        # conv_1_x_n_3
         # generator = ConvSettings(args, in_ch=1, out_ch=3, x_in=11, y_in=11, w_x=1, w_y=11, stride_x=1, stride_y=1,
         #                          pad=True, randmin=-2, randmax=2, outminrange=-127, outmaxrange=127)
 
         # conv_2
         # generator = ConvSettings(args, in_ch=2, out_ch=4, x_in=6, y_in=3, w_x=3, w_y=3, stride_x=1, stride_y=1,
         #                        pad=True, randmin=-2, randmax=2, outminrange=-126, outmaxrange=127)
 
+        # depthwise_2
+        # generator = ConvSettings(args, in_ch=3, out_ch=9, x_in=6, y_in=5, w_x=3, w_y=4, stride_x=2, stride_y=2,
+        #                         pad=True, randmin=-2, randmax=2, outminrange=-126, outmaxrange=127)
 
+        # depthwise_kernel_3x3
+        # generator = ConvSettings(args, in_ch=5, out_ch=5, x_in=4, y_in=5, w_x=3, w_y=3, stride_x=2, stride_y=2,
+        #                         pad=True, randmin=-2, randmax=2, outminrange=-126, outmaxrange=127)
+
+        # depthwise_eq_in_out_ch
+        generator = ConvSettings(args, in_ch=6, out_ch=6, x_in=4, y_in=5, w_x=2, w_y=3, stride_x=1, stride_y=1,
```

Comment:
Does the TC take care(error) of the case when out_channel is not a multiple of in_channel?

Reply:
No, will fix